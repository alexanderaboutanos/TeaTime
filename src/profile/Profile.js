/** @format */

import { useContext } from "react";
import UserContext from "../auth/UserContext";

const Profile = () => {
  const { currentUser } = useContext(UserContext);
  return (
    <div>
      <h2 style={{ margin: "25px" }}>My Account</h2>
      <p>
        <b>Username:</b> {currentUser.username}
      </p>
      <p>
        <b>First Name:</b> {currentUser.first_name}
      </p>
      <p>
        <b>Last Name:</b> {currentUser.last_name}
      </p>
    </div>
  );
};

export default Profile;
