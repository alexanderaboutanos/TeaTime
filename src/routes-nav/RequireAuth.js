/** @format */

import { useContext } from "react";
import { Navigate } from "react-router-dom";
import UserContext from "../auth/UserContext";

const RequireAuth = ({ children }) => {
  const { currentUser } = useContext(UserContext);

  console.debug("RequireAuth", "currentUser=", currentUser);

  return currentUser ? children : <Navigate to="/login" replace />;
};

export default RequireAuth;
