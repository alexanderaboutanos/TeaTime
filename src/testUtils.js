/** @format */

import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser1",
  first_name: "Test",
  last_name: "User",
};

const UserProvider = ({ children, currentUser = demoUser }) => (
  <UserContext.Provider value={{ currentUser }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
