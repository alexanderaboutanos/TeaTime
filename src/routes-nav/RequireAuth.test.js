/** @format */

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { UserProvider } from "../testUtils";
import Profile from "../profile/Profile";
import RequireAuth from "./RequireAuth";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <RequireAuth>
          <Profile />
        </RequireAuth>
      </UserProvider>
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <RequireAuth>
          <Profile />
        </RequireAuth>
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
