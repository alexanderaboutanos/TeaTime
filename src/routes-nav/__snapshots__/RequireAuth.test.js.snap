// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2
      style="margin: 25px;"
    >
      My Account
    </h2>
    <p>
      <b>
        Username:
      </b>
       testuser1
    </p>
    <p>
      <b>
        First Name:
      </b>
       Test
    </p>
    <p>
      <b>
        Last Name:
      </b>
       User
    </p>
  </div>
</DocumentFragment>
`;
